### PR TITLE
Add API Definition Validation

### PR DESCRIPTION
**Pull request by nicoburniske**
closes https://github.com/golemcloud/golem/issues/252

### CODE REVIEW

File: golem-worker-service-base/src/service/api_definition_validator.rs [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+            .map(|route| (route.binding.template.clone(), route))
+            .collect::<HashMap<_, _>>()
+            .into_values()
+            .map(|route| async move {
+                let id = &route.binding.template;
+                // TODO: Better error message.
+                self.template_service.get_latest(id).await.map_err(|e| {
+                    RouteValidationError::from_route(
+                        route,
+                        format!("Error getting latest template: {:?}", e),
```
Comment (justcoon) [inline golem-worker-service-base/src/service/api_definition_validator.rs:71]: i do not think is good idea to have full error there, internal errors should not leek there, 

i think for beginning we should just "Error getting latest template", later we can do it better

Comment (nicoburniske) [reply golem-worker-service-base/src/service/api_definition_validator.rs:71]: yeah I put the `TODO` there haha. I will remove the details and will revisit later

File: golem-worker-service-base/src/service/api_definition_validator.rs [hunk 2]
```diff
@@ -0,0 +1,158 @@
+use std::{collections::HashMap, sync::Arc};
+
+use crate::{
+    api_definition::{ApiDefinition, MethodPattern, PathPattern, Route},
+    service::template_service::TemplateService,
```
[the remark this answers is not in the document]
Comment (nicoburniske) [reply golem-worker-service-base/src/service/api_definition_validator.rs:5]: yeah that's a good point. we need to re-organize many of the files in worker-service-base in general. but I can make this change

**Review by afsalthaj — APPROVED** [review]

**Comment by afsalthaj** [comment]
LGTM. Let's get it merged.

**Review by justcoon — APPROVED** [review]

**Pull request merged**